### ISSUE TITLE
Increase the minimum sample size of bootstrapping

### ISSUE THREAD

**Issue by odashi** [neulab/ExplainaBoard#496]
TL;DR: increase the minimum sample size of bootstrapping to at least **1000**.

Confidence intervals are sometimes estimated from empirical distributions generated by bootstrapping. In `metric.py`, the sample size of each bootstrapping iteration is determined by the dataset size and the ratio `prop_samples`, and the minimum number of samples are currently settled to 1. In small datasets, the determined sample size becomes very small to estimate practical CIs.

Consider an example of estimated CIs on small/large datasets below. As shows in the figures (they show only the distributions of lower bounds), small sample sizes produce unpractical CIs because they tend to become very wide, even compared with the true STDs (1.0). It also shows that the tendency is independent from the dataset size, meaning that we need to choose larger sample sizes even when the dataset is small. According to the figures, 1000 is probably a better choice of the minimum sample size according to the tradeoff.

Preparation:
```python
import numpy as np
from matplotlib import pyplot as plt

def bootstrap(samples, iteration, sample_size, alpha):  
  resamples = np.random.choice(samples, size=(iteration, sample_size))
  means = resamples.mean(axis=1).sort()
  low_index = int(iteration * alpha / 2)
  high_index = int(iteration * (1.0 - alpha / 2))
  return means[low_index], means[high_index]

def show_lower_bound_distribution(dataset_size):
  samples = np.random.normal(size=(dataset_size,))
  labels = []
  low_dist = []
  for iteration in [10, 100, 1000]:
    for sample_size in [10, 100, 1000, 10000]:
      intervals = [bootstrap(samples, iteration, sample_size, 0.05) for _ in range(1000)]
      labels.append(f"iter={iteration}, samp={sample_size}")
      low_dist.append([x[0] for x in intervals])

  plt.boxplot(low_dist, labels=labels);
  plt.xticks(rotation=90);
```

Estimated CI lower bounds of a mall dataset:
```python
show_lower_bound_distribution(10)
```
![download](https://user-images.githubusercontent.com/1023695/190926774-41327cac-e922-4d4f-8601-56f20bee8b81.png)

Estimated CI lower bounds of a large dataset:
```python
show_lower_bound_distribution(1000)
```
![download](https://user-images.githubusercontent.com/1023695/190926649-e193e364-79ac-40dd-96e2-e1babd95d42c.png)
